props/saber_BC_buttons.h

```diff
@@ -442,91 +457,126 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
-          if (SFX_volmax) {
-            sound_library_.SayMaximumVolume();
-          } else {
-            beeper.Beep(0.5, 3000);
-          }
-          STDOUT.print("Maximum Volume \n");
-          max_vol_reached_ = true;
-        } else {
-          dynamic_mixer.set_volume(std::max<int>(VOLUME * 0.1,
-          dynamic_mixer.get_volume() - VOLUME * 0.90));
-          if (SFX_volmin) {
-            sound_library_.SayMininumVolume();
-          } else {
-            beeper.Beep(0.5, 1000);
-          }
-          STDOUT.print("Minimum Volume \n");
-          max_vol_reached_ = false;
-        }
-      #else
-        if (SFX_volmax) {
-          sound_library_.SayMaximumVolume();
-        } else {
-          beeper.Beep(0.5, 3000);
-        }
-        STDOUT.print("Maximum Volume \n");
-      #endif
+      QuickMaxVolume();
     }
   }
 
   void VolumeDown() {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    current_menu_angle_ = fusor.angle2();
     if (dynamic_mixer.get_volume() > (0.10 * VOLUME)) {
       dynamic_mixer.set_volume(std::max<int>(VOLUME * 0.1,
         dynamic_mixer.get_volume() - VOLUME * 0.10));
       if (SFX_voldown) {
-        sound_library_.SayVolumeDown();
+        hybrid_font.PlayPolyphonic(&SFX_voldown);
       } else {
         beeper.Beep(0.10, 2000);
         beeper.Beep(0.20, 1500);
       }
       STDOUT.print("Volume Down - Current Volume: ");
       STDOUT.println(dynamic_mixer.get_volume());
     } else {
-      #ifdef VOLUME_MENU_CYCLE
-        if (!min_vol_reached_) {
-          if (SFX_volmin) {
-            sound_library_.SayMininumVolume();
-          } else {
-            beeper.Beep(0.5, 1000);
-          }
-          STDOUT.print("Minimum Volume \n");
-          min_vol_reached_ = true;
-        } else {
-          dynamic_mixer.set_volume(VOLUME);
-          if (SFX_volmax) {
-            sound_library_.SayMaximumVolume();
-          } else {
-            beeper.Beep(0.5, 3000);
-          }
-          STDOUT.print("Maximum Volume \n");
-          min_vol_reached_ = false;
-        }
-      #else
-        if (SFX_volmin) {
-          sound_library_.SayMininumVolume();
-        } else {
-          beeper.Beep(0.5, 1000);
-        }
-        STDOUT.print("Minimum Volume \n");
-      #endif
+      QuickMinVolume();
+    }
+  }
+
+  void QuickMaxVolume() {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    dynamic_mixer.set_volume(VOLUME);
+    if (SFX_volmax) {
+      hybrid_font.PlayPolyphonic(&SFX_volmax);
+    } else {
+      beeper.Beep(0.5, 3000);
+    }
+    STDOUT.print("Maximum Volume \n");
+  }
+
+  void QuickMinVolume() {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    dynamic_mixer.set_volume(std::max<int>(VOLUME * 0.1,
+    dynamic_mixer.get_volume() - VOLUME * 0.90));
+    if (SFX_volmin) {
+      hybrid_font.PlayPolyphonic(&SFX_volmin);
+      } else {
+        beeper.Beep(0.5, 1000);
+      }
+      STDOUT.print("Minimum Volume \n");
+  }
+
+  void DetectMenuTurn() {
```

Comment:
It seems wasteful to have a separate function for detecting volume changes, isn't this mostly the same as the other menu detection code? Why have two?

Reply:
Sorry, I don't follow. Same as what other code, VolumeMenu() ?
VolumeMenu() just handles enter/exit.
DetectMenuTurn() is all motion stuff.
Did I double up on a function?